### ISSUE TITLE
example-web: cannot use @EnableProcessApplication

### ISSUE THREAD

**Issue by jangalinski** [camunda/camunda-bpm-spring-boot-starter#247]
When I comment in the annotation, the test fails with:

```

java.lang.RuntimeException: org.camunda.bpm.engine.ProcessEngineException: ENGINE-08043 Exception while performing 'Deployment of Process Application restApplication' => 'Start Process Application Service': Cannot register service org.camunda.bpm.platform.job-executor.process-application:type=restApplication with MBeans Container, service with same name already registered.
        at org.camunda.bpm.engine.spring.application.SpringProcessApplication.onApplicationEvent(SpringProcessApplication.java:99)
        at org.camunda.bpm.engine.spring.application.SpringProcessApplication.onApplicationEvent(SpringProcessApplication.java:48)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:167)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:383)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:337)
        at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:882)
        at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144
```

**Comment by osteinhauer** [comment]
Reproduced in the way that sometimes tests are ok and sometimes are failing.

Seems that ``@SpringBootTest`` doesn't recognized that different contexts are needed every time.

I tried to add

```java
...
@DirtiesContext
public class CamundaBpmRestIT {
...
}
```
and tests seem to be stable.

@jangalinski would you like to test with ``@DirtiesContext``?

**Comment by jangalinski** [comment]
Perfect. Can confirm, that after adding DirtiesContext, the build and integration-tests pass.
So my fixes for #182 that lead to a rollback yesterday can be applied.
But since I saw that Christian is working on the release of 2.1.2 right now, I'll wait til I re-push. 

Consider this closed.
